streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamSlidingWindowAggregate.java

```diff
@@ -311,13 +322,8 @@ private void putAndForward(final Window window,
             if (windowEnd > closeTime) {
                 //get aggregate from existing window
                 final Agg oldAgg = getValueOrNull(valueAndTime);
```

Comment:
Same here, can this ever be null? It doesn't seem like it, even if it's a new window we still pass in the initializer value so `valueAndTime` won't ever be null. ...right?

Reply:
I just ran the tests and it's _mostly_ true that it won't be null. We do have a test, and so it seems like a real life scenario (?), where the reduce initializer is null, so when creating a new window `valueAndTime` is null.

Reply:
True, I guess there's no reason the initializer can't return null. Nevermind then